rtos/TARGET_CORTEX/mbed_rtx_conf.h

```diff
@@ -40,7 +40,7 @@
 #endif
 
 // Increase the idle thread stack size when tickless is enabled
-#if defined(MBED_TICKLESS) && defined(LPTICKER_DELAY_TICKS) && (LPTICKER_DELAY_TICKS > 0)
```

Comment:
This change will affect all targets. C++ lp ticker wrapper is designed for all targets which have problem with back-to-back writes to timer compare register or have defined min cycles before match condition (not only STM has such problem, also: `ARM_MUSCA_A1_NS, NUMAKER_PFM_NUC472, etc.` ). `idle-thread-stack-size-tickless-extra` is defined for this case according to the description: `Additional size to add to the idle thread when tickless is enabled and LPTICKER_DELAY_TICKS is used`. 
I think this code cannot be removed.

Reply:
What I thought is that TICKLESS in itself implies stack increase, but maybe I'm wrong ...
In the case of ST targets with recent changes we will not have (LPTICKER_DELAY_TICKS > 0) and still we need stack size increase because the stack usage is increased in the driver layer.

Any idea how to cope with this without impacting other targets ?

Reply:
According to the description, `idle-thread-stack-size-tickless-extra` is designed for the case when both features are enabled:
- `TICKLESS`,
- `LPTICKER_DELAY_TICKS`

This means that the current code is consistent with the documentation.

Now we have the case when ST targets are not using `LPTICKER_DELAY_TICKS` because lp ticker issue is solved on HAL driver layer, but this solution still requires extra idle thread stack. This is the third case when the extra stack is required. 

I think that we need to start by changing the description here:
https://github.com/ARMmbed/mbed-os/blob/bf78dc44419c90c899b28d3ee225d96c0c1f15ef/rtos/mbed_lib.json#L21-L23
So the extra stack is not limited only to this one case (`TICKLESS + LPTICKER_DELAY_TICKS`) and then add another control macro like `EXTRA_IDLE_STACK_REQIRED` which would be automatically added in `TICKLESS + LPTICKER_DELAY_TICKS` configuration, but also there would be an option to define this macro directly in `targets.json`. Then the condition may have the following form:
```
// Increase the idle thread stack size when tickless is enabled
#if defined(EXTRA_IDLE_STACK_REQIRED)
#define EXTRA_IDLE_STACK MBED_CONF_RTOS_IDLE_THREAD_STACK_SIZE_TICKLESS_EXTRA
#else
#define EXTRA_IDLE_STACK 0
#endif
```

But I think that such changes should be approved by someone with better knowledge about the Mbed config @0xc0170.